tcl/org.testeditor.tcl.dsl.tests/src/org/testeditor/tcl/dsl/jvmmodel/AbstractTclGeneratorIntegrationTest.xtend

```diff
@@ -77,16 +76,13 @@ abstract class AbstractTclGeneratorIntegrationTest extends AbstractTclTest {
 		return fsa.allFiles.get(key)
 	}
 
-	/**
-	 * register the given model with the resource set (allows resolving links within same set)
-	 * @param fileExtension to be used (e.g. aml, tcl, tml, tsl)
-	 */
-	protected def <T extends EObject> T register(T model, String fileExtension) {
-		val uri = URI.createURI(UUID.randomUUID.toString + "." + fileExtension)
-
-		val newResource = resourceSet.createResource(uri)
-		newResource.contents.add(model)
-		return model
+	protected def <T extends EObject> T addToResourceSet(T model){
+		switch(model) {
+			case AmlModel: return model.addToResourceSet(resourceSet, "aml")
+			case TmlModel: return model.addToResourceSet(resourceSet, "tml")
+			case TslModel: return model.addToResourceSet(resourceSet, "tsl")
+			default: return  model.addToResourceSet(resourceSet, "tcl")
```

Comment:
I would explicitly list `TclModel` here and throw an exception in the default case.